### PR TITLE
Set `TypedOpParameter` to be `HONOR_PARAMETER` by default

### PR DESCRIPTION
**Pull request by lziq**
*Changes in this PR:*
1. Set `TypedOpParameter.HONOR_PARAMETER` to be default instead in `CompileOption`. 
2. Modified some test suites so `LEGACY` or `HONOR_PARAMETER` is explicitly specified. 
3. Renamed `resolveArithmeticOverflow` method to `checkIntegerOverflow` in `EvaluatingComiler.kt` since the method only checks integer overflow. Also, now it checks `TypedOpBehavior` instead of `TypingMode`, since the semantically, `TypingMode` is not really relevant.

### CODE REVIEW

File: lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt
Comment (lziq) [inline lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt:256]: See https://github.com/partiql/partiql-lang-kotlin/issues/784

File: testscript/pts/test-scripts/cast/cast-to-int.sqlts
Comment (lziq) [inline testscript/pts/test-scripts/cast/cast-to-int.sqlts:39]: The `HONOR_PARAMETER` mode rounds half to even, https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/ExprValueExtensions.kt#L348

**Comment by lziq** [comment]
Hold off the work of setting `TypedOpParameter.HONOR_PARAMETERS` as default until next PartiQL release, since we want to mark `TypedOpParameter.LEGACY` as deprecated first.

File: lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt
Comment (lziq) [inline lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt:256]: A bug for `TypedOpBehavior.HONOR_PARAMETERS`. The fix is not covered in this PR.

File: test/partiql-testscript/pts/test-scripts/cast/cast-to-int.sqlts
Comment (lziq) [inline test/partiql-testscript/pts/test-scripts/cast/cast-to-int.sqlts:39]: The `HONOR_PARAMETER` mode rounds half to even, https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/ExprValueExtensions.kt#L348

**Comment by lziq** [comment]
Unblock this PR, since Partiql 0.8 has been release.

File: lang/src/org/partiql/lang/eval/physical/PhysicalPlanCompilerImpl.kt
Comment (alancai98) [inline lang/src/org/partiql/lang/eval/physical/PhysicalPlanCompilerImpl.kt:388]: Similar comment to `EvaluatingCompiler.kt`. Thought this PR's change was only related to `typedOpBehavior` and not `typingMode`.

File: lang/test/org/partiql/lang/eval/CompOptions.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/CompOptions.kt:42]: Previous usage of this enum haven't been deleted/edited in some kdoc. E.g. https://github.com/partiql/partiql-lang-kotlin/blob/6c9949c0ac26806b20cc5dd42619b81fa29b099a/lang/test/org/partiql/lang/eval/EvaluatingCompilerIsTests.kt#L15-L17

File: lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt:254]: Was this test previously failing in this PR? Why is it being commented out in this PR?

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt
Comment (lziq) [reply lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt:528]: First of all, I think we want to always check if the result integer value overflows or not. It shouldn't be a configurable option. 

The reason why I want to change `TypingMode` to `TypedOpBehavior` is that integer constraints are modeled as type parameters currently, https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/types/StaticType.kt#L379 (even though I don't think this is a right way to model integer types). So the integer overflow checking semantically lines up better with whether type parameters work (`TypedOpBehavior`), instead of whether we should return `NULL`, `MISSING` or throw error (TypingMode).

File: lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt
Comment (lziq) [reply lang/test/org/partiql/lang/eval/EvaluatingCompilerIntTest.kt:254]: It works previously, since the default `LEGACY` mode `CAST` supports evaluating extremely small numbers. However, in the `HONOR_PARAMETERS` mode, it throws error when processing extremely small numbers. 

As what I tested, PostgreSQL does not allow processing extremely small numbers. I don't know if we want to support it or not. So I created another issue here, https://github.com/partiql/partiql-lang-kotlin/issues/784

File: lang/test/org/partiql/lang/eval/CompOptions.kt
Comment (lziq) [reply lang/test/org/partiql/lang/eval/CompOptions.kt:42]: Will update it

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt
Comment (alancai98) [reply lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt:528]: Could we save this change for a separate discussion and/or PR? I don't want to conflate changing the default `TypedOpBehavior` with a possibly breaking behavioral change.

I generally agree that the `TypingMode` should only really dictate the data type mismatches behaviors for clauses & operators. I think the prior `when` expression branching was only to verify that the overflow checking was done in `PERMISSIVE` mode because the static typing was only done in `PERMISSIVE` mode and not yet in `LEGACY` yet. See comments: https://github.com/partiql/partiql-lang-kotlin/blob/6c9949c0ac26806b20cc5dd42619b81fa29b099a/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L520-L533

That's why I don't think you can do a 1:1 drop-in replacement of `TypingMode.LEGACY` for `TypedOpBehavior.LEGACY`.

**Review by alancai98 — APPROVED** [review]

**Pull request merged**
